.github/workflows/ci.yml

```diff
@@ -245,15 +247,77 @@ jobs:
-          cat ${{ matrix.name }}-bench.csv
+            case "${{ matrix.host }}" in
+              "arm-linux-gnueabihf")
+              qemu-arm -E LD_LIBRARY_PATH=/usr/arm-linux-gnueabihf/lib/ /usr/arm-linux-gnueabihf/lib/ld-linux-armhf.so.3 src/bench/bench_dogecoin > ${{ matrix.name }}-bench.csv
+              cat ${{ matrix.name }}-bench.csv
+              ;;
+              "aarch64-linux-gnu")
+              qemu-aarch64 -E LD_LIBRARY_PATH=/usr/aarch64-linux-gnu/lib/ /usr/aarch64-linux-gnu/lib/ld-linux-aarch64.so.1 src/bench/bench_dogecoin > ${{ matrix.name }}-bench.csv
+              cat ${{ matrix.name }}-bench.csv
+              ;;
+              "x86_64-w64-mingw32")
+              make check $MAKEJOBS VERBOSE=1
+              ./src/bench/bench_dogecoin > ${{ matrix.name }}-bench.csv
+              cat ${{ matrix.name }}-bench.csv
+              ;;
+              "i686-w64-mingw32")
+              make check $MAKEJOBS VERBOSE=1
+              ./src/bench/bench_dogecoin > ${{ matrix.name }}-bench.csv
+              cat ${{ matrix.name }}-bench.csv
+              ;;
+              "x86_64-apple-darwin11")
+              make check $MAKEJOBS VERBOSE=1
+              ./src/bench/bench_dogecoin > ${{ matrix.name }}-bench.csv
+              cat ${{ matrix.name }}-bench.csv
+              ;;
+              "x86_64-unknown-linux-gnu")
+              make check $MAKEJOBS VERBOSE=1
+              ./src/bench/bench_dogecoin > ${{ matrix.name }}-bench.csv
+              cat ${{ matrix.name }}-bench.csv
+              ;;
+              "i686-pc-linux-gnu")
+              make check $MAKEJOBS VERBOSE=1
+              ./src/bench/bench_dogecoin > ${{ matrix.name }}-bench.csv
+              cat ${{ matrix.name }}-bench.csv
+              ;;
+            esac
 
       - name: Run tests
         if: ${{ matrix.run-tests }}
         run: |
-          make check $MAKEJOBS VERBOSE=1
-          qa/pull-tester/install-deps.sh
-          qa/pull-tester/rpc-tests.py --coverage
+            case "${{ matrix.host }}" in
+              "arm-linux-gnueabihf")
+              qemu-arm -E LD_LIBRARY_PATH=/usr/arm-linux-gnueabihf/lib/ /usr/arm-linux-gnueabihf/lib/ld-linux-armhf.so.3 src/test/test_dogecoin
+              ;;
+              "aarch64-linux-gnu")
+              qemu-aarch64 -E LD_LIBRARY_PATH=/usr/aarch64-linux-gnu/lib/ /usr/aarch64-linux-gnu/lib/ld-linux-aarch64.so.1 src/test/test_dogecoin
+              ;;
+              "x86_64-w64-mingw32")
+              make check $MAKEJOBS VERBOSE=1
+              qa/pull-tester/install-deps.sh
+              qa/pull-tester/rpc-tests.py --coverage
+              ;;
+              "i686-w64-mingw32")
+              make check $MAKEJOBS VERBOSE=1
+              qa/pull-tester/install-deps.sh
+              qa/pull-tester/rpc-tests.py --coverage
+              ;;
+              "x86_64-apple-darwin11")
+              make check $MAKEJOBS VERBOSE=1
+              ./src/test/test_dogecoin
+              ;;
```

Comment:
probably want to keep the macOS one out for the time being

Reply:
Awesome!  I concur with your recommendations.  I can merge your commits into my branch if that's alright?  We'll continue to work on automated testing for macos.

Reply:
> I can merge your commits into my branch if that's alright?

Of course!